<!DOCTYPE html>
<html lang="en">

<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="">
  <meta name="generator" content="Hugo 0.40" />

  <title>Matched tumor/normal pairs--a use case for the GenomicDataCommons Bioconductor package &middot; seandavi(s12)</title>

  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pure/1.0.0/pure-min.css">

  <!--[if lte IE 8]>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pure/1.0.0/grids-responsive-old-ie-min.css">
  <![endif]-->
  <!--[if gt IE 8]><!-->
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pure/1.0.0/grids-responsive-min.css">
  <!--<![endif]-->

  <!--[if lte IE 8]>
  <link rel="stylesheet" href="/css/side-menu-old-ie.css">
  <![endif]-->
  <!--[if gt IE 8]><!-->
  <link rel="stylesheet" href="/css/side-menu.css">
  <!--<![endif]-->

  <link rel="stylesheet" href="/css/blackburn.css">

  
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css">

  
  <link href="https://fonts.googleapis.com/css?family=Raleway" rel="stylesheet" type="text/css">

  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

 
  

  
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/zenburn.min.css">
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/yaml.min.js"></script>
  
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/r.min.js"></script>
  
  <script>hljs.initHighlightingOnLoad();</script>
  

  <link rel="shortcut icon" href="https://seandavi.github.io/img/favicon.ico" type="image/x-icon" />

  
  

</head>


<body>
<div id="layout">

  
<a href="#menu" id="menuLink" class="menu-link">
  
  <span></span>
</a>
<div id="menu">

  
  <a class="pure-menu-heading brand" href="/">seandavi(s12)</a>


  <div class="pure-menu">
    <ul class="pure-menu-list">
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="/post/"><i class='fa fa-list fa-fw'></i>Posts</a>
      
        </li>
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="/talk/"><i class='fa fa-microphone fa-fw'></i>Talks</a>
      
        </li>
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="/about/"><i class='fa fa-user fa-fw'></i>About</a>
      
        </li>
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="/publication/"><i class='fa fa-file-text fa-fw'></i>Publications</a>
      
        </li>
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="/"><i class='fa fa-home fa-fw'></i>Home</a>
      
        </li>
      
    </ul>
  </div>

  <div class="pure-menu social">
  <ul class="pure-menu-list">
      
    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="https://scholar.google.com/citations?user=hLFc29kAAAAJ"><i class="fa fa-graduation-cap fa-fw"></i>Scholar</a>
    </li>
    

    

    

    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="https://twitter.com/seandavis12" target="_blank"><i class="fa fa-twitter-square fa-fw"></i>Twitter</a>
    </li>
    

    

    

    

    

    

    

    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="https://github.com/seandavi" target="_blank"><i class="fa fa-github-square fa-fw"></i>GitHub</a>
    </li>
    

    

    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="https://bitbucket.org/seandavi" target="_blank"><i class="fa fa-bitbucket-square fa-fw"></i>Bitbucket</a>
    </li>
    

    

    


    

    

    

    

    

    

    

    

    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="http://slideshare.net/seandavi" target="_blank"><i class="fa fa-slideshare fa-fw"></i>SlideShare</a>
    </li>
    

    

    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="https://linkedin.com/in/seandavi" target="_blank"><i class="fa fa-linkedin-square fa-fw"></i>LinkedIn</a>
    </li>
    

    

    

    

    


    

    

    

    

    

    

    

    

  </ul>
</div>


  <div>
  <div class="small-print">
    <small>&copy; 2017 Sean Davis</small>
  </div>
  <div class="small-print">
  </div>
</div>

</div>


  <div id="main">


<div class="header">
  <h1>Matched tumor/normal pairs--a use case for the GenomicDataCommons Bioconductor package</h1>
  <h2></h2>
</div>
<div class="content">

  <div class="post-meta">

  <div>
    <i class="fa fa-calendar fa-fw"></i>
    <time>04 Mar 2017</time>
  </div>

  

  

  
  
  
  <div>
    <i class="fa fa-tags fa-fw"></i>
    
      <a class="post-taxonomy-tag" href="https://seandavi.github.io/tags/bioconductor">Bioconductor</a>&nbsp;&#47;
    
      <a class="post-taxonomy-tag" href="https://seandavi.github.io/tags/genomicdatacommons">GenomicDataCommons</a>&nbsp;&#47;
    
      <a class="post-taxonomy-tag" href="https://seandavi.github.io/tags/software">Software</a>&nbsp;&#47;
    
      <a class="post-taxonomy-tag" href="https://seandavi.github.io/tags/data-science">Data Science</a>&nbsp;&#47;
    
      <a class="post-taxonomy-tag" href="https://seandavi.github.io/tags/api">API</a>&nbsp;&#47;
    
      <a class="post-taxonomy-tag" href="https://seandavi.github.io/tags/r">R</a>
    
  </div>
  
  

</div>

  <div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The NCI Genomic Data Commons (<a href="https://gdc.nci.nih.gov">GDC</a>) is a reboot of the approach that NCI uses to manage and expose genomic and associated clinical and experimental metadata. I have been working on a <a href="https://bioconductor.org">Bioconductor</a> package that interfaces with the <a href="https://gdc-api.nci.nih.gov">GDC API</a> to provide search and data retrieval from within R.</p>
<div id="testing" class="section level2">
<h2>testing</h2>
<p>In the first of what will likely be a set of use cases for the <a href="https://github.com/Bioconductor/GenomicDataCommons">GenomicDataCommons</a>, I am going to address a question that came up on twitter from <a href="https://twitter.com/sleight82">@sleight82</a></p>
<blockquote>
<p><a href="https://twitter.com/sleight82/status/837898737540198400">Asking for a non-tweeter: can you find matched control samples?</a></p>
</blockquote>
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">asking for a non-tweeter: can you find matched control samples?</p>&mdash; Kenneth Daily (@sleight82) <a href="https://twitter.com/sleight82/status/837898737540198400?ref_src=twsrc%5Etfw">March 4, 2017</a></blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

<p>The answer is, “Yes.” I am going to assume that what the “non-tweeter” friend meant was that he or she was interested in finding matched tumor/normal data related to, for example, gene expression, and that the interest is in a specific disease category or TCGA. So, I am going to answer the more specific question:</p>
<blockquote>
<p>Can you find matched primary tumor/solid tissue normal samples and associated RNA-Seq gene expression files from TCGA breast cancer cases?</p>
</blockquote>
<pre class="r"><code>library(GenomicDataCommons)</code></pre>
<pre><code>## Loading required package: magrittr</code></pre>
<pre><code>## 
## Attaching package: &#39;GenomicDataCommons&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:GenomicDataCommons&#39;:
## 
##     count, filter, select</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<p>The GDC API puts some constraints on what can be done directly. But, since we are working in R, we have a toolbox that allows us to build a solution out of component parts. The strategy that I am going to employ is a three-step approach</p>
<ol style="list-style-type: decimal">
<li>Find RNA-Seq gene expression files derived from “Solid Tissue Normal”</li>
<li>Find RNA-Seq gene expression files derived from “Primary Tumor”</li>
<li>Limit cases from #1 and #2 that have gene expression results from both normal and tumor.</li>
</ol>
<p>In this code block, find all “HTSeq - Counts” files that are derived from “Solid Tissue Normal” in the project “TCGA-BRCA”. I used a combination of <code>files() %&gt;% select(available_fields('files') %&gt;% results()</code> to get an overview of the data available in the <code>files()</code> endpoint, followed by <code>grep_fields()</code> and <code>available_values()</code> to determine how to build filters.</p>
<pre class="r"><code>nl_ge_files = files() %&gt;%
    GenomicDataCommons::filter(~   cases.samples.sample_type==&#39;Solid Tissue Normal&#39; &amp;
               cases.project.project_id == &#39;TCGA-BRCA&#39; &amp;
               analysis.workflow_type == &quot;HTSeq - Counts&quot;) %&gt;%
    expand(c(&#39;cases&#39;,&#39;cases.samples&#39;)) %&gt;%
    results_all() %&gt;%
    as_tibble()</code></pre>
<p>And do the same, but now looking for gene expression from tumors.</p>
<pre class="r"><code>tm_ge_files = files() %&gt;%
    GenomicDataCommons::filter(~   cases.samples.sample_type==&#39;Primary Tumor&#39; &amp;
               cases.project.project_id == &#39;TCGA-BRCA&#39; &amp;
               analysis.workflow_type == &quot;HTSeq - Counts&quot;) %&gt;%
    expand(c(&#39;cases&#39;,&#39;cases.samples&#39;)) %&gt;%
    results_all() %&gt;%
    as_tibble()</code></pre>
<p>Now, we have two data frames describing the normal- and tumor-derived TCGA-BRCA gene expression files. Note that I asked the GDC to provide <code>cases.case_id</code> as part of the record using <code>select()</code>. By looking for the intersection of cases between these two sets of files, we can find cases that contain files derived from both tumor and normal tissue.</p>
<pre class="r"><code>nl_cases = bind_rows(nl_ge_files$cases, .id=&#39;file_id&#39;)
tm_cases = bind_rows(tm_ge_files$cases, .id=&#39;file_id&#39;)
matchedcases = intersect(nl_cases$case_id,
                         tm_cases$case_id)
# how many matched cases?
length(matchedcases)</code></pre>
<pre><code>## [1] 112</code></pre>
<p>We can now create a <code>data.frame</code> that contains file information for only the matched samples. Note
that the names of the matched cases are the file ids.</p>
<pre class="r"><code>matched_nl_files = nl_cases[nl_cases$case_id %in% matchedcases, &#39;file_id&#39;]
matched_tm_files = tm_cases[tm_cases$case_id %in% matchedcases, &#39;file_id&#39;]

matched_tn_ge_file_info = rbind(subset(nl_ge_files,file_id %in% matched_nl_files),
                                subset(tm_ge_files,file_id %in% matched_tm_files))
head(matched_tn_ge_file_info)</code></pre>
<pre><code>## # A tibble: 6 x 18
##   data_type  updated_datetime  file_name  submitter_id  file_id  file_size
##   &lt;chr&gt;      &lt;chr&gt;             &lt;chr&gt;      &lt;chr&gt;         &lt;chr&gt;        &lt;int&gt;
## 1 Gene Expr… 2017-03-04T16:39… f50438de-… f50438de-b31… ebd3bfb…    253804
## 2 Gene Expr… 2017-03-04T16:39… edea4e5b-… edea4e5b-ad4… 59602a3…    259917
## 3 Gene Expr… 2017-03-04T16:39… 60cc2580-… 60cc2580-b21… b7fda5e…    261904
## 4 Gene Expr… 2017-03-04T16:39… b504f2d6-… b504f2d6-335… 1e97e83…    251243
## 5 Gene Expr… 2017-03-04T16:39… 9570c964-… 9570c964-8d2… a03b240…    254621
## 6 Gene Expr… 2017-03-04T16:39… a0c9781b-… a0c9781b-0c1… f173ccd…    252823
## # ... with 12 more variables: cases &lt;list&gt;, id &lt;chr&gt;,
## #   created_datetime &lt;chr&gt;, md5sum &lt;chr&gt;, data_format &lt;chr&gt;, acl &lt;list&gt;,
## #   access &lt;chr&gt;, state &lt;chr&gt;, data_category &lt;chr&gt;, type &lt;chr&gt;,
## #   file_state &lt;chr&gt;, experimental_strategy &lt;chr&gt;</code></pre>
<p>Since the gene expression data are not that big, we can use the GDC API to download the data files directly. The GenomicDataCommons uses a cache for files, so the first time the code
below is run, data will be downloaded. After that, if the cache is in place, the cached
files will be used.</p>
<pre class="r"><code>fnames = lapply(as.character(matched_tn_ge_file_info$file_id),
                  function(file_id) {
                      gdcdata(file_id, progress = FALSE)
                  })</code></pre>
<p>Now, <code>fnames</code> should be a list of file names that can be read into and processed with R.</p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.5.0 RC (2018-04-16 r74624)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Sierra 10.12.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] dplyr_0.7.4              GenomicDataCommons_1.4.1
## [3] magrittr_1.5            
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.16                pillar_1.2.2               
##  [3] bindr_0.1.1                 compiler_3.5.0             
##  [5] GenomeInfoDb_1.15.5         XVector_0.19.9             
##  [7] bitops_1.0-6                tools_3.5.0                
##  [9] zlibbioc_1.25.0             digest_0.6.15              
## [11] jsonlite_1.5                tibble_1.4.2               
## [13] evaluate_0.10.1             lattice_0.20-35            
## [15] pkgconfig_2.0.1             rlang_0.2.0                
## [17] Matrix_1.2-14               cli_1.0.0                  
## [19] DelayedArray_0.5.35         curl_3.2                   
## [21] yaml_2.1.19                 parallel_3.5.0             
## [23] blogdown_0.6                xfun_0.1                   
## [25] bindrcpp_0.2.2              GenomeInfoDbData_1.1.0     
## [27] xml2_1.2.0                  httr_1.3.1                 
## [29] stringr_1.3.1               knitr_1.20                 
## [31] hms_0.4.2                   rappdirs_0.3.1             
## [33] S4Vectors_0.17.43           IRanges_2.13.29            
## [35] stats4_3.5.0                rprojroot_1.3-2            
## [37] grid_3.5.0                  glue_1.2.0                 
## [39] Biobase_2.39.2              R6_2.2.2                   
## [41] BiocParallel_1.13.3         rmarkdown_1.9              
## [43] bookdown_0.7                readr_1.1.1                
## [45] backports_1.1.2             htmltools_0.3.6            
## [47] matrixStats_0.53.1          BiocGenerics_0.25.3        
## [49] GenomicRanges_1.31.23       assertthat_0.2.0           
## [51] SummarizedExperiment_1.9.18 utf8_1.1.3                 
## [53] stringi_1.2.2               lazyeval_0.2.1             
## [55] RCurl_1.95-4.10             crayon_1.3.4</code></pre>
</div>
</div>


  
<div class="prev-next-post pure-g">
  <div class="pure-u-1-24" style="text-align: left;">
    
    <a href="https://seandavi.github.io/post/2015/07/hello-r-markdown/"><i class="fa fa-chevron-left"></i></a>
    
  </div>
  <div class="pure-u-10-24">
    
    <nav class="prev">
      <a href="https://seandavi.github.io/post/2015/07/hello-r-markdown/">Hello R Markdown</a>
    </nav>
    
  </div>
  <div class="pure-u-2-24">
    &nbsp;
  </div>
  <div class="pure-u-10-24">
    
    <nav class="next">
      <a href="https://seandavi.github.io/post/2017/11/agricultural-genomics-may-benefit-from-human-genomic-data-and-software-engineering/">Agricultural genomics may benefit from human genomic data and software engineering</a>
    </nav>
    
  </div>
  <div class="pure-u-1-24" style="text-align: right;">
    
    <a href="https://seandavi.github.io/post/2017/11/agricultural-genomics-may-benefit-from-human-genomic-data-and-software-engineering/"><i class="fa fa-chevron-right"></i></a>
    
  </div>
</div>



  
<div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    if (window.location.hostname == "localhost")
        return;

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'seandavi-github-io';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


</div>

</div>
</div>
<script src="https://seandavi.github.io/js/ui.js"></script>


<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-93043521-1', 'auto');
  ga('send', 'pageview');

</script>





</body>
</html>

